lib/active_fedora/noid/minter/db.rb

```diff
@@ -0,0 +1,36 @@
+require 'noid'
+
+module ActiveFedora
+  module Noid
+    module Minter
+      class Db < Base
+
+        protected
+
+        # Uses pessimistic lock to ensure the record fetched is the same one updated.
+        # Should be fast enough to avoid terrible deadlock.
+        # Must lock because of multi-connection context! (transaction is per connection -- not enough)
+        def next_id
+          id = ''
+          MinterState.with_lock do
+            state = MinterState.first || do
```

Comment:
Obviously.  But there should be *no* callers outside the lock.  The best way to ensure that is not to break it out.